core/src/map.cpp

```diff
@@ -216,30 +224,28 @@ void Map::loadSceneAsync(const char* _scenePath, bool _useScenePosition,
         nextScene = impl->nextScene;
     }
 
-    runAsyncTask([nextScene, _onMapReady, _onMapReadyUserData, _sceneUpdates, _onSceneUpdateError, this](){
+    runAsyncTask([nextScene, _onSceneReady, _sceneUpdates, this](){
 
-            bool newSceneLoaded = SceneLoader::loadScene(platform, nextScene, _sceneUpdates, _onSceneUpdateError);
+            bool newSceneLoaded = SceneLoader::loadScene(platform, nextScene, _sceneUpdates);
 
-            impl->jobQueue.add([nextScene, newSceneLoaded, _onMapReady, _onMapReadyUserData, _onSceneUpdateError, this]() {
+            impl->jobQueue.add([nextScene, newSceneLoaded, _onSceneReady, this]() {
                     {
                         std::lock_guard<std::mutex> lock(impl->sceneMutex);
                         if (nextScene == impl->nextScene) {
                             impl->nextScene.reset();
                         } else {
                             // loadScene[Async] was called in the meantime.
+                            if (_onSceneReady) { _onSceneReady(false, {}); }
```

Comment:
Now that I've removed the use of global refs for the updateErrorCallback it's not really needed anymore. Previously calling scene-update from Android created a global ref that had to be released in the SceneErrorCallback function. There was a bug that the ref was not released when the scene-update succeeded.  https://github.com/tangrams/tangram-es/pull/1526/files#diff-3d0c7fc4582b39c4afec2835dcd73f43L134